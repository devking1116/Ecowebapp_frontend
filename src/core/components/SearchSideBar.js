/* @flow */
import React from 'react';
import reactCSS from 'reactcss';

import SideBar from './SideBar';
import SearchSideBarHeader from './SearchSideBarHeader';

type Props = {
  title: string,
  catigories?: Array<{
    catigoriesId: string,
    label: string,
  }>,
  expandedWidth: number | string,
  isExpanded?: bool,
  hiddenWhenClosed?: bool,
  expandSideBar: () => void,
  collapseSideBar: () => void,
  children?: any,
};

/**
 * The text search filter and results.
 */
export default function SearchSideBar(props: Props): React.Element<*> {
  const {
    title,
    catigories,
    expandedWidth,
    isExpanded,
    hiddenWhenClosed,
    expandSideBar,
    collapseSideBar,
    children,
    ...otherProps,
  } = props;

  const styles = reactCSS({
    'default': {
      sideBar: {
        backgroundColor: '#f8f8f8',
      },
    },
  });

  return (
    <SideBar
      header={
        <SearchSideBarHeader
          title={title}
          catigories={catigories}
          expandSideBar={expandSideBar}
          collapseSideBar={collapseSideBar}
        />
      }
      expandedWidth={expandedWidth}
      isExpanded={isExpanded}
      hiddenWhenClosed={hiddenWhenClosed}
      style={styles.sideBar}
      {...otherProps}
    >
  {isExpanded?
    children:
    null
  }
    </SideBar>
  );
}
