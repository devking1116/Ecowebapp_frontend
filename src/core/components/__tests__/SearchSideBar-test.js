import React from 'react';
import renderer from 'react-test-renderer';

import SearchSideBar from '../SearchSideBar';

test('the title is displayed in the header', () => {
  const component = renderer.create(
    <SearchSideBar title="Test Title" isExpanded>
      <i>Test Child 0</i>
      <i>Test Child 1</i>
      <i>Test Child 2</i>
    </SearchSideBar>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('the catigories selector is displayed in the header, if catigories are given', () => {
  const component = renderer.create(
    <SearchSideBar catigories={[ 'Test catigories'/* //TODO, relpace with real catigories */ ]} isExpanded>
      <i>Test Child 0</i>
      <i>Test Child 1</i>
      <i>Test Child 2</i>
    </SearchSideBar>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('when expanded, it displays its children in a single column at the given width.', () => {
  const component = renderer.create(
    <SearchSideBar expandedWidth={200} isExpanded>
      <i>Test Child 0</i>
      <i>Test Child 1</i>
      <i>Test Child 2</i>
    </SearchSideBar>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('when not expaned, it does not display its children.', () => {
  const component = renderer.create(
    <SearchSideBar>
      <i>Test Child 0</i>
      <i>Test Child 1</i>
      <i>Test Child 2</i>
    </SearchSideBar>
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('when expanded, it gives its children the isExpanded prop.', () => {
  const component = renderer.create(
    <SearchSideBar isExpanded>
      <i>Test Child 0</i>
      <i>Test Child 1</i>
      <i>Test Child 2</i>
    </SearchSideBar>
  );
  const tree = component.toJSON();
  const [ _, ...children ] = tree.children;

  children.forEach(({ children: [ child ]}) => {
    expect(child.props.isExpanded).toBe(true);
  });
});

test('when not expanded, it does not gives its children the isExpanded prop.', () => {
  const component = renderer.create(
    <SearchSideBar>
      <i>Test Child 0</i>
      <i>Test Child 1</i>
      <i>Test Child 2</i>
    </SearchSideBar>
  );
  const tree = component.toJSON();
  const [ _, ...children ] = tree.children;

  children.forEach(({ children: [ child ]}) => {
    expect(child.props.isExpanded).toBeFalsy();
  });
});
