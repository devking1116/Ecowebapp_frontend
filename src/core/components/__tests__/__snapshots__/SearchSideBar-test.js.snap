exports[`test the catigories selector is displayed in the header, if catigories are given 1`] = `
<div
  className="ui vertical menu"
  style={
    Object {
      "backgroundColor": "#f8f8f8",
      "height": "100%",
      "position": "relative",
      "width": undefined,
    }
  }>
  <div
    className="item"
    onClick={[Function]}>
    <div
      style={
        Object {
          "WebkitJustifyContent": "space-between",
          "display": "flex",
          "height": 20,
          "justifyContent": "space-between",
        }
      }>
      <h3
        style={
          Object {
            "color": "#5e656a",
          }
        } />
      <div>
        NYI: catigorie selector
      </div>
      <i
        aria-hidden="true"
        className="grey minus circle large inverted icon"
        onClick={undefined} />
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <i
      isExpanded={true}>
      Test Child 0
    </i>
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <i
      isExpanded={true}>
      Test Child 1
    </i>
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <i
      isExpanded={true}>
      Test Child 2
    </i>
  </div>
</div>
`;

exports[`test the title is displayed in the header 1`] = `
<div
  className="ui vertical menu"
  style={
    Object {
      "backgroundColor": "#f8f8f8",
      "height": "100%",
      "position": "relative",
      "width": undefined,
    }
  }>
  <div
    className="item"
    onClick={[Function]}>
    <div
      style={
        Object {
          "WebkitJustifyContent": "space-between",
          "display": "flex",
          "height": 20,
          "justifyContent": "space-between",
        }
      }>
      <h3
        style={
          Object {
            "color": "#5e656a",
          }
        }>
        Test Title
      </h3>
      <i
        aria-hidden="true"
        className="grey minus circle large inverted icon"
        onClick={undefined} />
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <i
      isExpanded={true}>
      Test Child 0
    </i>
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <i
      isExpanded={true}>
      Test Child 1
    </i>
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <i
      isExpanded={true}>
      Test Child 2
    </i>
  </div>
</div>
`;

exports[`test when expanded, it displays its children in a single column at the given width. 1`] = `
<div
  className="ui vertical menu"
  style={
    Object {
      "backgroundColor": "#f8f8f8",
      "height": "100%",
      "position": "relative",
      "width": 200,
    }
  }>
  <div
    className="item"
    onClick={[Function]}>
    <div
      style={
        Object {
          "WebkitJustifyContent": "space-between",
          "display": "flex",
          "height": 20,
          "justifyContent": "space-between",
        }
      }>
      <h3
        style={
          Object {
            "color": "#5e656a",
          }
        } />
      <i
        aria-hidden="true"
        className="grey minus circle large inverted icon"
        onClick={undefined} />
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <i
      isExpanded={true}>
      Test Child 0
    </i>
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <i
      isExpanded={true}>
      Test Child 1
    </i>
  </div>
  <div
    className="item"
    onClick={[Function]}>
    <i
      isExpanded={true}>
      Test Child 2
    </i>
  </div>
</div>
`;

exports[`test when not expaned, it does not display its children. 1`] = `
<div
  className="ui vertical icon menu"
  style={
    Object {
      "backgroundColor": "#f8f8f8",
      "display": undefined,
      "height": "100%",
      "position": "relative",
      "width": undefined,
    }
  }>
  <div
    className="item"
    onClick={[Function]}>
    <div
      style={
        Object {
          "WebkitJustifyContent": "flex-end",
          "display": "flex",
          "height": 20,
          "justifyContent": "flex-end",
        }
      }>
      <i
        aria-hidden="true"
        className="content large icon"
        onClick={undefined} />
    </div>
  </div>
</div>
`;
